extensions/wikia/PhalanxII/maintenance/MigratePhalanxUserBlocks.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+require_once __DIR__ . '/../../../../maintenance/Maintenance.php';
+
+/**
+ * Cleanup infinite Phalanx username blocks
+ *
+ * Script to close the accounts for permanent Phalanx username blocks
+ * older than three months and delete the relevant Phalanx block.
+ */
+class MigratePhalanxUserBlocks extends Maintenance {
+	private $batchSize = 1000;
+	private $curBlockId = 0;
+	private $dryRun = false;
+
+	public function __construct() {
+		parent::__construct();
+		$this->addOption( 'dry-run', "Perform a dry run, don't close any accounts" );
+	}
+
+	public function execute() {
+		$this->dryRun = $this->getOption( 'dry-run', false );
+
+		$this->setCurrentUser();
+
+		do {
+			$userBlocks = $this->getUserBlocks();
+
+			if ( empty( $userBlocks ) ) {
+				break;
+			}
+
+			$blockIdsClosed = $this->closeAccounts( $userBlocks );
+
+			$this->deletePhalanxBlocks( $blockIdsClosed );
+
+			$this->output( count( $blockIdsClosed ) . " user accounts closed.\n" );
```

Comment:
I'm using the last block ID processed to get the next batch of 1000 blocks to process so I don't need to wait for the slaves to catch up, as we would never be selecting the same group of blocks anyway. Would the wait still be necessary in this case?